ddl/db_integration_test.go

```diff
@@ -3162,6 +3162,165 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
+	err = tk.ExecToErr("drop global temporary table xxx")
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+
+	// testing for drop table if exists which is not global temporary
+	err = tk.ExecToErr("drop global temporary table if exists tb")
+	c.Assert(core.ErrDropTableOnTemporaryTable.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop global temporary table if exists ltemp1")
+	c.Assert(core.ErrDropTableOnTemporaryTable.Equal(err), IsTrue)
+	tk.MustExec("drop global temporary table if exists xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.xxx'"))
+	err = tk.ExecToErr("drop global temporary table if exists xxx,tb")
+	c.Assert(core.ErrDropTableOnTemporaryTable.Equal(err), IsTrue)
+	err = tk.ExecToErr("drop global temporary table if exists test.tb")
+	c.Assert(core.ErrDropTableOnTemporaryTable.Equal(err), IsTrue)
+
+	// testing for drop global temporary table successfully
+	tk.MustExec("drop global temporary table temp")
+	err = tk.ExecToErr("select * from temp")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+
+	tk.MustExec("drop global temporary table test.temp1")
+	err = tk.ExecToErr("select * from temp2")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+
+	tk.MustExec("create global temporary table temp (id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1 (id int) on commit delete rows")
+
+	tk.MustExec("drop global temporary table temp, temp1")
+	err = tk.ExecToErr("select * from temp")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+	err = tk.ExecToErr("select * from temp1")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+
+	tk.MustExec("create global temporary table temp (id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1 (id int) on commit delete rows")
+
+	tk.MustExec("drop global temporary table if exists temp")
+	tk.MustQuery("show warnings").Check(testkit.Rows())
+	err = tk.ExecToErr("select * from temp")
+	c.Assert(infoschema.ErrTableNotExists.Equal(err), IsTrue)
+}
+
+func (s *testIntegrationSuite3) TestDropLocalTemporaryTableWithDropTemporaryStmt(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("set tidb_enable_noop_functions=true")
+	tk.MustExec("set tidb_enable_global_temporary_table=true")
+	clearSQL := "drop table if exists tb, tb2, temp, temp1, ltemp1, ltemp2, testt.ltemp3"
+	tk.MustExec(clearSQL)
+	defer tk.MustExec(clearSQL)
+	// two normal table test.tb, test.tb2, a temporary table with name test.tb2
+	tk.MustExec("create table tb(id int)")
+	tk.MustExec("create table tb2(id int)")
+	tk.MustExec("insert into tb2 values(1)")
+	tk.MustExec("create temporary table tb2(id int)")
+	// two global temporary table test.temp, test.temp1
+	tk.MustExec("create global temporary table temp(id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1(id int) on commit delete rows")
```

Comment:
From the test function name `TestDropLocalTemporaryTableWithDropTemporaryStmt`  this is a bit confusing

Reply:
This is copied, so..